pkg/trace/agent/agent.go

```diff
@@ -422,6 +426,21 @@ func traceContainsError(trace pb.Trace) bool {
 	return false
 }
 
+func filteredByTags(root *pb.Span, require, reject []*config.Tag) bool {
+	for _, tag := range reject {
+		if v, ok := root.Meta[tag.K]; ok && (tag.V == "" || (v != "" && v == tag.V)) {
+			return true
+		}
+	}
+	for _, tag := range require {
+		v, ok := root.Meta[tag.K]
+		if !ok || (tag.V != "" && v != "" && v != tag.V) {
```

Comment:
```suggestion
		if !ok || (tag.V != "" && v != tag.V) {
```
Same here, this should be good enough.